Detectors/HMPID/calibration/testWorkflow/HMPIDDCSDataProcessorSpec.h

```diff
@@ -0,0 +1,315 @@
+  {
+    // fill CCDB with RefIndex (std::vector<TF1> arrMean)
+    const auto& payload = mProcessor->getRefIndexObj();
+    auto& info = mProcessor->getccdbRefInfo();
+
+    auto image = o2::ccdb::CcdbApi::createObjectImage(&payload, &info);
+    LOG(info) << "Sending object " << info.getPath() << "/"
+              << info.getFileName() << " of size " << image->size()
+              << " bytes, valid for " << info.getStartValidityTimestamp()
+              << " : " << info.getEndValidityTimestamp();
+
+    output.snapshot(
+      Output{o2::calibration::Utils::gDataOriginCDBPayload, "RefIndex", 0},
+      *image.get());
+    output.snapshot(
+      Output{o2::calibration::Utils::gDataOriginCDBWrapper, "RefIndex", 0},
+      info);
+  }
+
+  std::vector<std::string> aliases = {
+    "HMP_ENV_PENV",
+    "HMP_MP[0..6]_GAS_PMWPC",
+    "HMP_MP[0..6]_LIQ_LOOP_RAD_[0..2]_IN_TEMP",
+    "HMP_MP[0..6]_LIQ_LOOP_RAD_[0..2]_OUT_TEMP",
+    "HMP_MP_[0..6]_SEC_[0..5]_HV_VMON",
+    "HMP_TRANPLANT_MEASURE_[0..29]_WAVELENGHT",
+    "HMP_TRANPLANT_MEASURE_[0..29]_ARGONREFERENCE",
+    "HMP_TRANPLANT_MEASURE_[0..29]_ARGONCELL",
+    "HMP_TRANPLANT_MEASURE_[0..29]_C6F14REFERENCE",
+    "HMP_TRANPLANT_MEASURE_[0..29]_C6F14CELL"};
+
+  bool isRunStarted = false;
+  bool mCheckRunStartStop = false;
+  o2::dcs::RunStatusChecker mRunChecker{o2::detectors::DetID::getMask("HMP")};
+
+  std::unique_ptr<HMPIDDCSProcessor> mProcessor;
+  HighResClock::time_point mTimer;
+
+}; // end class HMPIDDCSDataProcessor
+} // namespace hmpid
+
+namespace framework
+{
+
+o2::framework::DataProcessorSpec getHMPIDDCSDataProcessorSpec()
+{
+
+  using clbUtils = o2::calibration::Utils;
+
+  std::vector<OutputSpec> outputs;
+  outputs.emplace_back(
+    ConcreteDataTypeMatcher{o2::calibration::Utils::gDataOriginCDBPayload,
+                            "ChargeCut"},
+    Lifetime::Sporadic);
+  outputs.emplace_back(
+    ConcreteDataTypeMatcher{o2::calibration::Utils::gDataOriginCDBWrapper,
+                            "ChargeCut"},
+    Lifetime::Sporadic);
+
+  outputs.emplace_back(
+    ConcreteDataTypeMatcher{o2::calibration::Utils::gDataOriginCDBPayload,
+                            "RefIndex"},
+    Lifetime::Sporadic);
+  outputs.emplace_back(
+    ConcreteDataTypeMatcher{o2::calibration::Utils::gDataOriginCDBWrapper,
+                            "RefIndex"},
+    Lifetime::Sporadic);
+
+  return o2::framework::DataProcessorSpec{
+    "hmp-dcs-data-processor", Inputs{{"input", "DCS", "HMPDATAPOINTS"}},
+    outputs, AlgorithmSpec{adaptFromTask<o2::hmpid::HMPIDDCSDataProcessor>()},
+    Options{{"ccdb-path",
+             VariantType::String,
+             "localhost:8080",
+             {"Path to CCDB"}},
+            {"use-ccdb-to-configure",
+             VariantType::Bool,
+             false,
+             {"Use CCDB to configure"}},
+            {"follow-hmpid-run",
+             VariantType::Bool,
+             false,
```

Comment:
if I get it right, with default setting (no run following) you will not produce any output? Is this what you want?

Reply:
Yes, this is what we want.

Reply:
ok, but then it is not clear why this is an option, moreover set to false by default ?

Reply:
You are right, the idea was to optimize the code later on! Anyway, since it is simple to modify it, we are doing to do it now.